tests/e2e/p/l1.go

```diff
@@ -0,0 +1,146 @@
+
+package p
+
+import (
+	"math"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/api/info"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/tests/fixture/e2e"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
+	"github.com/ava-labs/avalanchego/vms/example/xsvm/genesis"
+	"github.com/ava-labs/avalanchego/vms/platformvm"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+)
+
+var _ = e2e.DescribePChain("[L1]", func() {
+	tc := e2e.NewTestContext()
+	require := require.New(tc)
+
+	ginkgo.It("creates and updates L1 validators", func() {
+		env := e2e.GetEnv(tc)
+		nodeURI := env.GetRandomNodeURI()
+
+		tc.By("verifying Etna is activated", func() {
+			infoClient := info.NewClient(nodeURI.URI)
+			upgrades, err := infoClient.Upgrades(tc.DefaultContext())
+			require.NoError(err)
+
+			now := time.Now()
+			if !upgrades.IsEtnaActivated(now) {
+				ginkgo.Skip("Etna is not activated. L1s are enabled post-Etna, skipping test.")
+			}
+		})
+
+		tc.By("loading the wallet")
+		var (
+			keychain   = env.NewKeychain()
+			baseWallet = e2e.NewWallet(tc, keychain, nodeURI)
+			pWallet    = baseWallet.P()
+			pClient    = platformvm.NewClient(nodeURI.URI)
+			owner      = &secp256k1fx.OutputOwners{
+				Threshold: 1,
+				Addrs: []ids.ShortID{
+					keychain.Keys[0].Address(),
+				},
+			}
+		)
+
+		tc.By("creating the chain genesis")
+		genesisKey, err := secp256k1.NewPrivateKey()
+		require.NoError(err)
+
+		genesisBytes, err := genesis.Codec.Marshal(genesis.CodecVersion, &genesis.Genesis{
+			Timestamp: time.Now().Unix(),
+			Allocations: []genesis.Allocation{
+				{
+					Address: genesisKey.Address(),
+					Balance: math.MaxUint64,
+				},
+			},
+		})
+		require.NoError(err)
+
+		var subnetID ids.ID
+		tc.By("issuing a CreateSubnetTx", func() {
+			subnetTx, err := pWallet.IssueCreateSubnetTx(
+				owner,
+				tc.WithDefaultContext(),
+			)
+			require.NoError(err)
+
+			subnetID = subnetTx.ID()
+		})
+
+		tc.By("verifying a Permissioned Subnet was successfully created", func() {
+			require.NotEqual(constants.PrimaryNetworkID, subnetID)
+
+			subnet, err := pClient.GetSubnet(tc.DefaultContext(), subnetID)
+			require.NoError(err)
+			require.Equal(
+				platformvm.GetSubnetClientResponse{
+					IsPermissioned: true,
+					ControlKeys: []ids.ShortID{
+						keychain.Keys[0].Address(),
+					},
+					Threshold: 1,
+				},
+				subnet,
+			)
+		})
+
+		var chainID ids.ID
+		tc.By("issuing a CreateChainTx", func() {
+			chainTx, err := pWallet.IssueCreateChainTx(
+				subnetID,
+				genesisBytes,
+				constants.XSVMID,
+				nil,
+				"No Permissions",
+				tc.WithDefaultContext(),
+			)
+			require.NoError(err)
+
+			chainID = chainTx.ID()
+		})
+
+		address := []byte{}
+		tc.By("issuing a ConvertSubnetTx", func() {
+			_, err := pWallet.IssueConvertSubnetTx(
```

Comment:
(No action needed) Curious/Surprised that `IssueConvertSubnetTx` doesn't take the initial L1 validators...is this just because the test is for the old spec that included `RecoverSubnetTx`?

Reply:
Correct. This PR was factored out of #3397 which adds the initial L1 validators.